dts/bindings/pinctrl/hpmicro,hpm-pinctrl.yaml

```diff
@@ -0,0 +1,103 @@
+# Copyright 2022 hpmicro
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  The HPMicro pin controller is a singleton node responsible for
+  controlling pin function selection and pin properties. For example, you can
+  use this node to route UART0 RX to pin PA10 and enable the pull-up resistor
+  on the pin. Remapping is also supported.
+
+  The node has the 'pinctrl' node label set in your SoC's devicetree,
+  so you can modify it like this:
+
+    &pinctrl {
+            /* your modifications go here */
+    };
+
+  All device pin configurations should be placed in child nodes of the
+  'pinctrl' node, as shown in this example:
+
+  &pinctrl {
+    pinmux_uart0: pinmux_uart0 {
+      group0 {
+        pinmux = <HPMICRO_PINMUX(HPMICRO_PIN(HPMICRO_PORTY, 7), 0, 2)>,
+        <HPMICRO_PINMUX(HPMICRO_PIN(HPMICRO_PORTY, 6), 0, 2)>;
+      };
+    };
+    pinmux_gpiob: pinmux_gpiob {
+      group0 {
+        pinmux = <HPMICRO_PINMUX(HPMICRO_PIN(HPMICRO_PORTB, 18), 0, 0)>,
+            <HPMICRO_PINMUX(HPMICRO_PIN(HPMICRO_PORTB, 19), 0, 0)>,
+            <HPMICRO_PINMUX(HPMICRO_PIN(HPMICRO_PORTB, 20), 0, 0)>;
+            bias-pull-down;
+      };
+    };
+  };
+
+  A group can also specify shared pin properties common to all the specified
+  pins, such as the 'bias-pull-up' property in group 2. Here is a list of
+  supported standard pin properties:
+
+  - drive-open-drain: Open-drain drive mode.
+  - bias-disable: Disable pull-up/down (default, not required).
+  - bias-pull-up: Enable pull-up resistor.
+  - bias-pull-down: Enable pull-down resistor.
+  - input-enable: enable input.
+  - input-schmitt-enable: enable input schmitt circuit.
+
+  The macro HPMICRO_PINMUX sets the pinmux function:
+  - The first parameter is the pin number, the macro HPMICRO_PIN generates pin numbers.
+  - The second parameter enables to indicate that the pin is an analogue pin.
+  - The third parameter is the remapping number,
+    look up the datasheet to obtain the parameters.
+
+compatible: "hpmicro,hpm-pinctrl"
+
+include:
+  - name: base.yaml
+  - name: pincfg-node-group.yaml
+    child-binding:
+      child-binding:
+        property-allowlist:
+          - drive-open-drain
+          - input-enable
+          - bias-disable
+          - bias-pull-down
+          - bias-pull-up
+          - input-schmitt-enable
+
+child-binding:
+  description: HPMicro pin controller pin group
+  child-binding:
+    description: |
+      HPMicro pin controller pin configuration node.
+    properties:
+      pinmux:
+        required: true
+        type: array
+        description: |
+          Represents pin mux settings. Use HPMICRO_PINMUX
+          With:
+          - pin: The gpio pin number (0, 1, ..., GPIO_NUM_MAX)
+          - analog: bool, select analog pin in pad
+          - alt_select: alt0-alt31
+      drive-strength:
```

Comment:
This doesn't look right.  I'm assuming these are binary values for the register field or something.  Don't these values mean something?  How would anyone know how to set this field?